### ISSUE TITLE
[Enhancement] Use paginated requests

### ISSUE THREAD

**Issue by Timtam** [rasmuslos/ShelfPlayer#167]
Hello,

my library is so big (about 8000 audio books, but several hundred thousand files) that ABS takes a while to load the entire library. So long in fact that ShelfPlayer stops trying and tells me that i'm offline, and the entire ABS instance becomes unuseable until the process finished. Plus I have to increase my node RAM size to even process the request. The web app doesn't have this problem since it limits the sample size and automatically loads new items in the background when the user scrolls further down.
I'd be great to implement this in ShelfPlayer too, to save some bandwidth and actually allow me to use my server with it ;).

**Comment by rasmuslos** [comment]
ShelfPlayer does paginate fetch requests. At what point does the request time out?

**Comment by gh0sti** [comment]
> Hello,
> 
> 
> 
> my library is so big (about 8000 audio books, but several hundred thousand files) that ABS takes a while to load the entire library. So long in fact that ShelfPlayer stops trying and tells me that i'm offline, and the entire ABS instance becomes unuseable until the process finished. Plus I have to increase my node RAM size to even process the request. The web app doesn't have this problem since it limits the sample size and automatically loads new items in the background when the user scrolls further down.
> 
> I'd be great to implement this in ShelfPlayer too, to save some bandwidth and actually allow me to use my server with it ;).

Would it be possible to shrink it down by converting your mp3 to m4b one file for each of your books? ABS has that function built in to convert mp3 multiple files into 1 m4b

**Comment by Timtam** [comment]
@rasmuslos It happens on the screen that pops up sometimes directly after starting the app, it displays a message saying "Synchronizing progress" (or something like that), which stays open for roughly 15 seconds before switching to the offline screen.

**Comment by Timtam** [comment]
@gh0sti probably, but it'd also exclude any other players that access my audio book library through other means, like WebDAV, FTP etc. Considered it, ditched it long ago. Thanks anyway.

**Comment by rasmuslos** [comment]
I don’t think that logging in is the issue, it’s the Spotlight index. It isn’t paginated. I disable it for large libraries until it is.

**Comment by Timtam** [comment]
Did this get pushed as an official release yet? I tried multiple times today and it still doesn't work :(. Is there anything more I can do to help?

**Comment by Timtam** [comment]
Also worth mentioning, the screen I was talking about above disappeared, but it just shows a loading screen forever. sometimes it manages to load the library once, but stops/hangs as soon as I switch tabs to e.g. the library tab.
